tests/backup/backup_helper.go

```diff
@@ -1254,6 +1292,213 @@ func IsBackupLocationPresent(bkpLocation string, ctx context.Context, orgID stri
+}
+
+// GetOrdinalScheduleBackupName returns the name of the schedule backup at the specified ordinal position for the given schedule
+func GetOrdinalScheduleBackupName(ctx context.Context, scheduleName string, ordinal int, orgID string) (string, error) {
+	allScheduleBackupNames, err := Inst().Backup.GetAllScheduleBackupNames(ctx, scheduleName, orgID)
+	if err != nil {
+		return "", err
+	}
+	if len(allScheduleBackupNames) == 0 {
+		return "", fmt.Errorf("no backups were found for the schedule [%s]", scheduleName)
+	}
+	if ordinal < 1 {
+		return "", fmt.Errorf("the ordinal value [%d] for schedule backups with schedule name [%s] is invalid. valid values range from 1 to [%d]", ordinal, scheduleName, len(allScheduleBackupNames))
+	}
+	if ordinal > len(allScheduleBackupNames) {
+		return "", fmt.Errorf("schedule backups with schedule name [%s] have not been created up to the provided ordinal value [%d]", scheduleName, ordinal)
+	}
+	return allScheduleBackupNames[ordinal-1], nil
+}
+
+// GetOrdinalScheduleBackupNameAfterCheck returns the name of the schedule backup at the specified ordinal position for the given schedule after success check
+func GetOrdinalScheduleBackupNameAfterCheck(ctx context.Context, scheduleName string, ordinal int, orgID string) (string, error) {
+	ordinalScheduleBackupName, err := task.DoRetryWithTimeout(func() (interface{}, bool, error) {
+		ordinalScheduleBackupName, err := GetOrdinalScheduleBackupName(ctx, scheduleName, ordinal, orgID)
+		if err != nil {
+			if ordinal < 1 {
+				return "", false, err
+			}
+			return "", true, err
+		}
+		return ordinalScheduleBackupName, false, nil
+	}, maxWaitPeriodForBackupCompletionInMinutes*time.Minute, 30*time.Second)
+	if err != nil {
+		return "", err
+	}
+	_, err = backupSuccessCheck(ordinalScheduleBackupName.(string), orgID, 0, 0, ctx)
+	if err != nil {
+		return "", err
+	}
+	return ordinalScheduleBackupName.(string), nil
+}
+
+// GetFirstScheduleBackupName returns the name of the first schedule backup for the given schedule
+func GetFirstScheduleBackupName(ctx context.Context, scheduleName string, orgID string) (string, error) {
+	allScheduleBackupNames, err := Inst().Backup.GetAllScheduleBackupNames(ctx, scheduleName, orgID)
+	if err != nil {
+		return "", err
+	}
+	if len(allScheduleBackupNames) == 0 {
+		return "", fmt.Errorf("no backups found for schedule %s", scheduleName)
+	}
+	return allScheduleBackupNames[0], nil
+}
+
+// GetFirstScheduleBackupNameAfterCheck returns the name of the first schedule backup for the given schedule after success check
+func GetFirstScheduleBackupNameAfterCheck(ctx context.Context, scheduleName string, orgID string) (string, error) {
+	firstScheduleBackupName, err := GetFirstScheduleBackupName(ctx, scheduleName, orgID)
+	if err != nil {
+		return "", err
+	}
+	_, err = backupSuccessCheck(firstScheduleBackupName, orgID, 0, 0, ctx)
+	if err != nil {
+		return "", err
+	}
+	return firstScheduleBackupName, nil
+}
```

Comment:
Same here. Not required.

Reply:
I have implemented GetFirstScheduleBackupNameAfterCheck to complement CreateScheduleBackupWithoutCheck.

Reply:
Removed GetFirstScheduleBackupNameAfterCheck.